src/Faker/ORM/Doctrine/Populator.php

```diff
@@ -68,13 +84,24 @@ public function execute($entityManager = null)
             throw new \InvalidArgumentException("No entity manager passed to Doctrine Populator.");
         }
 
+        $batchCountIterateCount = 0;
```

Comment:
Maybe we can even inline the variable `$batchCountIterateCount` and replace it with a query instead?

```diff
diff --git a/src/Faker/ORM/Doctrine/Populator.php b/src/Faker/ORM/Doctrine/Populator.php
index 692486f..9087c63 100644
--- a/src/Faker/ORM/Doctrine/Populator.php
+++ b/src/Faker/ORM/Doctrine/Populator.php
@@ -84,22 +84,24 @@ class Populator
             throw new \InvalidArgumentException("No entity manager passed to Doctrine Populator.");
         }

-        $batchCountIterateCount = 0;
         $insertedEntities = array();
+
         foreach ($this->quantities as $class => $number) {
             $generateId = $this->generateId[$class];
+
             for ($i=0; $i < $number; $i++) {
-                $batchCountIterateCount++;
                 $insertedEntities[$class][]= $this->entities[$class]->execute(
                     $entityManager,
                     $insertedEntities,
                     $generateId
                 );
-                if ($batchCountIterateCount % $this->batchSize === 0) {
+
+                if (count($insertedEntities) % $this->batchSize === 0) {
                     $entityManager->flush();
                     $entityManager->clear($class);
                 }
             }
+
             $entityManager->flush();
             $entityManager->clear($class);
         }
```

Reply:
:+1: